packages/connector-chromium/src/connector.ts

```diff
@@ -0,0 +1,534 @@
+            element,
+            request: networkRequest,
+            resource: resourceUrl,
+            response: networkResponse
+        };
+
+        return data;
+    }
+
+    /** Returns the HTMLElement that initiated a request */
+    private getElementFromResponse(source: puppeteer.Response | puppeteer.Request): HTMLElement | null {
+        const request = 'request' in source ?
+            source.request() :
+            source;
+
+        const redirectChain = request.redirectChain();
+        const requestUrl = redirectChain && redirectChain.length > 0 ?
+            redirectChain[0].url() :
+            source.url();
+
+        /*
+         * TODO: Check what happens with prefetch, etc.
+         * `type` can be "parser", "script", "preload", and "other": https://chromedevtools.github.io/debugger-protocol-viewer/tot/Network/#type-Initiator
+         */
+        // The doesn't seem to be an initiator in puppeteer :/
+        if (this._dom && requestUrl.startsWith('http')) {
+            return getElementByUrl(this._dom, requestUrl, this._finalHref);
+        }
+
+        return null;
+    }
+
+    /**
+     * Manually download the site's favicon:
+     *
+     * * uses the `src` attribute of `<link rel="icon">` if present.
+     * * uses `favicon.ico` and the final url after redirects.
+     */
+    private async getFavicon(dom: HTMLDocument) {
+        const element = (await dom.querySelectorAll('link[rel~="icon"]'))[0];
+        const href = (element && element.getAttribute('href')) || '/favicon.ico';
+
+        try {
+            debug(`resource ${href} to be fetched`);
+            const fullFaviconUrl = this._finalHref + href.substr(1);
+
+            await this._engine.emitAsync('fetch::start', { resource: fullFaviconUrl });
+
+            const content = await this.fetchContent(new URL(fullFaviconUrl));
+
+            const data: FetchEnd = {
+                element: null,
+                request: content.request,
+                resource: content.response.url,
+                response: content.response
+            };
+
+            await this._engine.emitAsync('fetch::end::image', data);
+        } catch (error) {
+            const event: FetchError = {
+                element,
+                error,
+                hops: [],
+                resource: href
+            };
+
+            await this._engine.emitAsync('fetch::error', event);
+        }
+    }
+
+    private getRawResponse(response: puppeteer.Response) {
+        const connector = this; // eslint-disable-line
+
+        return async function (this: any) {
+
+            const that = this; // eslint-disable-line
+
+            if (that._rawResponse) {
+                return that._rawResponse;
+            }
+
+            const rawContent = await response.buffer();
+            const responseHeaders = response.headers();
+
+            if (rawContent && rawContent.length.toString() === responseHeaders['Content-Length']) {
```

Comment:
Just double-checking: is the casing of `'Content-Length'` guaranteed here?

Reply:
It used to be guaranteed in the `connector-chrome` but in this particular one it might not be. Let me check.